numpy/core/src/npysort/quicksort.c.src

```diff
@@ -198,6 +208,7 @@ stack_pop:
         }
         pr = *(--sptr);
         pl = *(--sptr);
+        cdepth = *(--psdepth);
```

Comment:
hm right I misread the break condition.
I still don't like it as much. The goto variant makes the code structure more similar to the recursive code which is much easier to understand, you just have to interpret the goto stack pop as the equivalent of the return statement in the recursive one.
But if you prefer it like that I can change it